### PR TITLE
Add Subject Selector Component

### PR DESCRIPTION
**Pull request by n-ong**
# Overview

This PR includes code for the Subject Selector / Dropdown menu. It includes a sorting util that will be used to sort the loaded subjects from the backend api. Still need to implement the component on the CourseSearchForm and load in data from the backend.

# Issues Addressed

#6

### CODE REVIEW

File: frontend/src/main/components/CourseSearch/SubjectSelector.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react'
+import { Form } from 'react-bootstrap';
+import { compareValues } from "main/utils/sortHelper"
+
+const SubjectSelector = ({ subjects, subject, setSubject, controlId, onChange = null, label = "Subject" }) => {
```
Comment (Peake2) [inline frontend/src/main/components/CourseSearch/SubjectSelector.js:5]: the subject prop is unused, should probably be removed

**Review by Peake2 — CHANGES_REQUESTED** [review]
One small change

File: frontend/src/main/components/CourseSearch/SubjectSelector.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/main/components/CourseSearch/SubjectSelector.js:5]: Agreed.

**Review by pconrad — CHANGES_REQUESTED** [review]
Agree with @Peake2 's review to consider removing the unused subject prop.

Other than that, LGTM... will likely approve once that change is made.

**Review by Peake2 — APPROVED** [review]
Review 2: LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**